### PR TITLE
feat: add ability to set timezone of an employee

### PR DESCRIPTION
**Pull request by djaiss**
This PR adds support for timezones.

You fool, don't forget these steps:

- [ ] Unit tests
- [ ] Tests with Cypress
- [ ] Documentation
- [ ] Dummy data

### CODE REVIEW

File: app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php [hunk 1]
```diff
@@ -65,6 +66,7 @@ private function save(array $data): void
         $this->employee->last_name = $data['last_name'];
         $this->employee->email = $data['email'];
         $this->employee->phone_number = $data['phone'];
+        $this->employee->timezone = $data['phone'];
```
Comment (asbiin) [inline app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php:69]: I think it's
```suggestion
        $this->employee->timezone = $data['timezone'];
```

File: app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ public function rules(): array
             'last_name' => 'required|string|max:255',
             'email' => 'required|email:rfc|max:255',
             'phone' => 'nullable|max:255',
+            'timezone' => 'required|integer',
```
Comment (asbiin) [inline app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php:31]: You should pass the timezone code itself, like 'UTC' or 'Europe/Paris'

**Review by asbiin — CHANGES_REQUESTED** [review]
I know it's just a draft, but I made some comments anyway ;)

File: app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php (hunk 1 above)
Comment (djaiss) [reply app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php]: Fixed it in another way.

File: app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php (hunk 2 above)
Comment (djaiss) [reply app/Services/Company/Employee/PersonalDetails/SetPersonalDetails.php]: Excellent idea, this is what i've done.

**Comment by djaiss** [comment]
@asbiin I think I'll keep this PR small.

This PR will become `add ability to set timezone of an employee`.

I'll make the changes of supporting timezones while displaying dates in another PR.

**Comment by djaiss** [comment]
Holy shit.

The `DateTimeZone` actually differs from platform to platform and from PHP versions too. Tests therefore, are unreliable.

I don't think we should use this class, and rather use a static list somewhere.

**Comment by djaiss** [comment]
I don't think it's that big a deal. We'll keep using the default timezones for now, provided by PHP.

**Pull request merged**